packages/vitest-pool-workers/src/pool/plugin.ts

```diff
@@ -0,0 +1,122 @@
+import crypto from "node:crypto";
+import fs from "node:fs/promises";
+import path from "node:path";
+import { cloudflarePool } from "./pool";
+import type { WorkersPoolOptions } from "./config";
+import type { inject } from "vitest";
+import type { Vite, VitestPluginContext } from "vitest/node";
+
+const cloudflareTestPath = path.resolve(
+	import.meta.dirname,
+	"../worker/lib/cloudflare/test.mjs"
+);
+
+export interface WorkerPoolOptionsContext {
+	// For accessing values from `globalSetup()` (e.g. ports servers started on)
+	// in Miniflare options (e.g. bindings, upstream, hyperdrives, ...)
+	inject: typeof inject;
+}
+
+function ensureArrayIncludes<T>(array: T[], items: T[]) {
+	for (const item of items) {
+		if (!array.includes(item)) {
+			array.push(item);
+		}
+	}
+}
+
+function ensureArrayExcludes<T>(array: T[], items: T[]) {
+	for (let i = 0; i < array.length; i++) {
+		if (items.includes(array[i])) {
+			array.splice(i, 1);
+			i--;
+		}
+	}
+}
+
+const requiredConditions = ["workerd", "worker", "module", "browser"];
+const requiredMainFields = ["browser", "module", "jsnext:main", "jsnext"];
+
+export function cloudflareTest(
+	options:
+		| WorkersPoolOptions
+		| ((
+				ctx: WorkerPoolOptionsContext
+		  ) => Promise<WorkersPoolOptions> | WorkersPoolOptions)
+): Vite.Plugin {
+	// Use a unique ID for each `cloudflare:test` module so updates in one `main`
+	// don't trigger re-runs in all other projects, just the one that changed.
+	const uuid = crypto.randomUUID();
+	let main: string | undefined;
+	return {
+		name: "@cloudflare/vitest-pool-workers",
+		api: {
+			setMain(newMain: string) {
+				main = newMain;
+			},
+		},
+		configureVitest(context: VitestPluginContext) {
```

Comment:
Annoyingly, in the monorepo it's not picking up the Vitest types for `Plugin` here, so this gives me type errors in my editor